File: data pendukung/soal1.pl
% Fakta gejala
 gejala(daun_menguning) :- ask('Apakah daun menguning? (y/n) ').
 gejala(bercak_hitam) :- ask('Apakah terdapat bercak hitam? (y/n) ').
 gejala(daun_berlubang) :- ask('Apakah daun berlubang? (y/n) ').
 gejala(tanaman_layu) :- ask('Apakah tanaman layu? (y/n) ').

 % Aturan penyakit berdasarkan logika proposisional
 penyakit(nematoda_akar) :-
     gejala(daun_menguning),
     gejala(tanaman_layu).

 penyakit(ulat_grayak) :-
     gejala(daun_berlubang),
     gejala(bercak_hitam).

 penyakit(jamur_daun) :-
     gejala(daun_menguning),
     gejala(bercak_hitam).

 penyakit(kumbang_pemakan_daun) :-
     gejala(daun_berlubang).

 penyakit(fusarium_sp) :-
     gejala(tanaman_layu),
     \+ gejala(bercak_hitam),
     \+ gejala(daun_berlubang).

 % Proses input dari user
 ask(Pertanyaan) :-
     write(Pertanyaan),
     read(Respon),
     Respon = y.

 % Program utama
 start :-
     findall(P, penyakit(P), List),
     (List = [] -> write('Tidak ada penyakit yang terdeteksi berdasarkan gejala tersebut.');
      write('Penyakit yang terdeteksi: '), nl, tampilkan(List)).

 % Menampilkan hasil
 tampilkan([]).
 tampilkan([H|T]) :-
     write('- '), write(H), nl,
     tampilkan(T).

File: soal1.py
def tanya_gejala(pertanyaan):
    jawaban = input(pertanyaan + " (y/n): ").lower()
    return jawaban == 'y'

def diagnosa():
    # Tanya gejala ke user
    daun_menguning = tanya_gejala("Apakah daun menguning?")
    bercak_hitam = tanya_gejala("Apakah terdapat bercak hitam?")
    daun_berlubang = tanya_gejala("Apakah daun berlubang?")
    tanaman_layu = tanya_gejala("Apakah tanaman layu?")

    # Daftar penyakit yang terdeteksi
    penyakit_terdeteksi = []

    # Cek logika aturan
    if daun_menguning and tanaman_layu:
        penyakit_terdeteksi.append("Nematoda akar")
    if daun_berlubang and bercak_hitam:
        penyakit_terdeteksi.append("Ulat grayak")
    if daun_menguning and bercak_hitam:
        penyakit_terdeteksi.append("Jamur daun")
    if daun_berlubang:
        penyakit_terdeteksi.append("Kumbang pemakan daun")
    if tanaman_layu and not bercak_hitam and not daun_berlubang:
        penyakit_terdeteksi.append("Fusarium sp.")

    # Tampilkan hasil
    if penyakit_terdeteksi:
        print("\nPenyakit yang terdeteksi:")
        for p in penyakit_terdeteksi:
            print(f"- {p}")
    else:
        print("\nTidak ada penyakit yang terdeteksi berdasarkan gejala tersebut")

if __name__ == "__main__":
    diagnosa()
